autopilot/autopilot.go

```diff
@@ -9,188 +9,89 @@ import (
-	RefreshHealth(ctx context.Context) error
-	Slab(ctx context.Context, key object.EncryptionKey) (object.Slab, error)
-	SlabsForMigration(ctx context.Context, healthCutoff float64, limit int) ([]api.UnhealthySlab, error)
-	DeleteHostSector(ctx context.Context, hk types.PublicKey, root types.Hash256) error
-	FetchPartialSlab(ctx context.Context, key object.EncryptionKey, offset, length uint32) ([]byte, error)
-
-	// scanner
-	ScanHost(ctx context.Context, hostKey types.PublicKey, timeout time.Duration) (resp api.HostScanResponse, err error)
-
-	// settings
-	GougingParams(ctx context.Context) (api.GougingParams, error)
 	GougingSettings(ctx context.Context) (gs api.GougingSettings, err error)
-	UploadSettings(ctx context.Context) (us api.UploadSettings, err error)
-
-	// syncer
-	SyncerPeers(ctx context.Context) (resp []string, err error)
-
-	// txpool
+	Hosts(ctx context.Context, opts api.HostOptions) ([]api.Host, error)
 	RecommendedFee(ctx context.Context) (types.Currency, error)
-	TransactionPool(ctx context.Context) (txns []types.Transaction, err error)
-
-	// wallet
+	ScanHost(ctx context.Context, hostKey types.PublicKey, timeout time.Duration) (api.HostScanResponse, error)
+	SyncerPeers(ctx context.Context) (resp []string, err error)
+	UploadSettings(ctx context.Context) (us api.UploadSettings, err error)
 	Wallet(ctx context.Context) (api.WalletResponse, error)
-	WalletPending(ctx context.Context) (resp []wallet.Event, err error)
-	WalletRedistribute(ctx context.Context, outputs int, amount types.Currency) (ids []types.TransactionID, err error)
 }
 
-type Autopilot struct {
-	alerts alerts.Alerter
-	bus    Bus
-	logger *zap.SugaredLogger
+type Autopilot interface {
+	Handler() http.Handler
+	Run()
+	Shutdown(context.Context) error
+	Trigger(forceScan bool) bool
+	Uptime() time.Duration
+}
+
+type autopilot struct {
+	b Bus
+	l *zap.SugaredLogger
 
-	c *contractor.Contractor
+	c contractor.Contractor
 	m migrator.Migrator
+	p pruner.Pruner
 	s scanner.Scanner
+	w wallet.Wallet
 
-	tickerDuration time.Duration
-	wg             sync.WaitGroup
+	hearbeat time.Duration
+	wg       sync.WaitGroup
 
 	startStopMu       sync.Mutex
 	startTime         time.Time
 	shutdownCtx       context.Context
 	shutdownCtxCancel context.CancelFunc
 	ticker            *time.Ticker
 	triggerChan       chan bool
-
-	mu               sync.Mutex
-	pruning          bool
-	pruningLastStart time.Time
-	pruningAlertIDs  map[types.FileContractID]types.Hash256
-
-	maintenanceTxnIDs []types.TransactionID
 }
 
 // New initializes an Autopilot.
-func New(cfg config.Autopilot, masterKey utils.MasterKey, bus Bus, logger *zap.Logger) (_ *Autopilot, err error) {
-	logger = logger.Named("autopilot")
+func New(ctx context.Context, cancel context.CancelFunc, b Bus, c contractor.Contractor, m migrator.Migrator, p pruner.Pruner, s scanner.Scanner, w wallet.Wallet, heartbeat time.Duration, logger *zap.Logger) Autopilot {
```

Comment:
Better to return the actual type here rather than an interface. Like how `LimitReader` returns a `*LimitedReader` even though it could also return a `io.Reader`. 
It's often times more useful because you might want to call a method on the concrete type before you pass it to some method. e.g. in testing

Same thing for the other submodules. Instead I'd export the actual type and move the interface to where we need it. Which for the autopilot's submodules should be the autopilot and the autopilot would probably go into the bus package.